src/bdecode.cpp

```diff
@@ -109,6 +109,14 @@ namespace libtorrent
 		// reading a key or a vale. 0 means key 1 is value
 		std::uint32_t state:1;
 	};
+
+	// diff between current and next item offset
+	// should only be called for non last item in array
+	int calc_token_in_array_size(bdecode_token const& t)
```

Comment:
how about ``token_source_span()``? something to indicate it's the number of bytes this token covers in the source encoding.